sdk/python/kfp/dsl/component_factory.py

```diff
@@ -132,6 +133,14 @@ def make_pip_install_command(
 PIP_DISABLE_PIP_VERSION_CHECK=1 {pip_install_commands} && "$0" "$@"
 '''
 
+# Creates and activates a virtual environment in a temporary directory.
+# The environment inherits the system site packages.
+_use_venv_script_template = '''
+tmp=$(mktemp -d)
```

Comment:
worth noting that `mktemp` is not posix compliant, 
https://stackoverflow.com/questions/2792675/how-portable-is-mktemp1

though it's pretty widespread

Reply:
What's the concern about not being POSIX compliant? Despite Linux, does the project support any other UNIX-like system?

The main executor also uses `mktemp`. We could look into platformdir or xdg base dir if needed.

Reply:
I don't have concern per se, just noting it here for posterity, if we are using this elsewhere, then the precedent is already set and it's out of scope of this PR imo